demos/portal-embed-demo/public/js/ttt/audio-controller.js

```diff
@@ -0,0 +1,369 @@
+     */
+    addPlaylist(id) {
+        this.playlist.push(id);
+        this._updateController();
+    }
+
+    /**
+     * Remove audio from playlist
+     * @param {String} id - audio ID
+     */
+    removePlaylist(id) {
+        // Simply create new playlist filtering the ID to remove
+        const newPlaylist = [];
+        this.playlist
+            .filter(audioId => audioId !== id)
+            .map(audioId => newPlaylist.push(audioId));
+        this.playlist = newPlaylist;
+
+        // Simply decrement the current index. There could be more 
+        // sophisticated UX to this but the quality of the audio 
+        // controller is not the point for this demo.
+        if(this.playlistIndex !== 0){
+            this.playlistIndex--;
+            this._createNewAudio(this.playlist[this.playlistIndex])
+            if(this._isPlaying()) {
+                // If the controller was playing an audio, 
+                // start playing the new audio
+                this.play();
+            }
+        }
+
+        this._updateController();
+    }
+
+    /**
+     * Hiding the controller with anmimation
+     */
+    hide() {
+        this.style.transition = 'bottom 0.6s';
+        this.style.bottom = '-80px';
+    }
+
+    /**
+     * Showing the controller with anmimation
+     */
+    show() {
+        this.style.transition = 'bottom 0.2s';
+        this.style.bottom = '0px';
+    }
+
+    /**
+     * Adding anything required to do on portal activate
+     */
+    handleActivation() {
+        if(this._isPlaying()) {
+            // change the left transition position
+            const currentLeft = this._getAxisCurrentLeft();
+            const left = this.timeline.offsetWidth;
+            const duration = this.durationSec * (1 - currentLeft/left);
+            this.axis.style.transition = `${duration}s linear`;
+            this.axis.style.left = `${left}px`;
+        }
+    }
+
+    /**
+     * Adding event listeners for the buttons
+     */
+    _hookEvents() {
+        this.playButton.addEventListener('click', evt => {
+            if(this._isPlaying()){
+                this.pause();
+                return;
+            }
+            this.play();
+        });
+        this.nextButton.addEventListener('click', evt => this.next());
+        this.prevButton.addEventListener('click', evt => this.prev());
+        this.audio.addEventListener("ended", evt => {
+            this.pause();
+            this._createNewAudio(this.playlist[this.playlistIndex]);
+       });
+    }
+
+    /**
+     * Creating a new audio element
+     * @param {String} id - audio ID
+     */
+    _createNewAudio(id) {
+
+        // Simply replace the exisiting audio element with a new one
+        if(this.root.querySelector('audio')){
+            this.root.removeChild(this.root.querySelector('audio'));
+        }
+        const audio = document.createElement('audio');
+        const source = document.createElement('source');
```

Comment:
Since we only ever have one source, it's simpler to just set `src` on the media element. (The HTML spec also recommends this approach as you can change what is playing on a media element by changing its `src`, but not by modifying its `<source>` children: https://html.spec.whatwg.org/#the-source-element)

Reply:
Understood. Thank you for this comment. Fixed it.
```javascript
audio.src = `/mp3/${id}.mp3`;
this.root.appendChild(audio);
```